backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs

```diff
@@ -58,7 +58,7 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
 
     public async Task<ICollection<DayCardDto>?> GetDayCardsOfWeekAsync(RequestDayCardDto requestDto)
     {
-        var mondayDate = requestDto.Date.AddDays(-(int)requestDto.Date.DayOfWeek).Date;
+        var mondayDate = requestDto.Date.AddDays(-((7 + (requestDto.Date.DayOfWeek - DayOfWeek.Monday)) % 7)).Date;
```

Comment:
Why you need 7 here?
7%7=0

Reply:
Sunday = 0 
Monday = 1
[src](https://learn.microsoft.com/en-us/dotnet/api/system.dayofweek?view=net-7.0)
If current daty is Sunday: (0 - 1) % 7 = -1
In my implementation: (7 + 0 - 1) % 7 = 6

Reply:
This statement looks strange, please simplify it